rust/src/meas/geometric/mod.tex

```diff
@@ -0,0 +1,283 @@
+    where $n$ is the dimension of $v,w$.
+    Here it's implicitly assumed that $|v|=|w|$.
+    
+    Let $s=$\texttt{ scale}, $[a,b]=$\texttt{ bounds},
+    $Y=\function(v)$, and $Z=\function(w)$.
+    Our goal is to show that \texttt{MaxDivergence}  $D_{\infty}(Y||Z)$, also defined in the proof definition document, adheres to the privacy guarantee.
+    
+    Because the \function is defined such that 
+    \texttt{sample\_two\_sided\_geometric}, it properly
+    simulates a random variable $Z \sim $ 
+    \texttt{shift} $+ 
+    \text{ Geo}$(\texttt{scale}), with
+    probability mass function
+    $f_Z(z)\propto \exp{-|z-\texttt{shift}|/\texttt{scale}}$, along each axis of the input vector, or in the linear domain, in the trivial case with 1-dimensional input.
+    So it follows that 
+     \[
+    D_{\infty}(Y||Z) = \max_{S \subseteq Supp(Y)}\left[\ln (\frac{\Pr[Y \in S]}{\Pr[Z \in S]})\right] 
+    \]
+    
+    \[
+    = \max_{y\in Supp(Y)}\left[\ln (\frac{\Pr[
+    \function(v) = y]}
+    {\Pr[\function(w) = y]})\right]
+    \]
+    
+    \[
+    = \max_{y\in Supp(Y)}\sum^n_{i=1}\left[\ln (\frac{\exp{-|y_i-v_i|/s}}
+    {\exp{-|y_i-w_i|/s}})\right]
+    \]
+    
+    \[
+    = \max_{y\in Supp(Y)}\sum^n_{i=1}\left[\ln (\exp{-(|y_i-v_i|-|y_i-w_i|)/s})\right]
+    \]
+    
+    \[\leq \max_{y\in Supp(Y)}\sum^n_{i=1}\Big|
+    |y_i-v_i|-|y_i-w_i|\Big|/s\]
+    
+    % \[\leq \max_{y,y'\in Supp(Y)}\sum^n_{i=1}\Big|
+    % |y_i-v_i|-|y_i-w_i|\Big|/s\]
+    
+    \[
+    \leq \sum^n_{i=1}|v_i-w_i|/s
+    =\frac{1}{s}\sum^n_{i=1}|v_i-w_i|
+    =\texttt{d\_in}/\texttt{scale}
+    \]
+    
+ Because \function() output is discrete, we are able to switch from set to point-wise comparison as in the first and second lines. The summation in the third line comes from the vector input, where dimension $n$ represents the size of the input. 
+ The exponential function inside the summation is given by the fact of the probability mass function of the geometric distribution 
+ brought up earlier. 
+After basic algebraic manipulation, we apply the triangle inequality and obtain the last line. The result can be written 
+as \texttt{d\_in}/\texttt{scale}, as desired.
+This completes the proof of privacy guarantee.
+ 
+ \textbf{Failure cases.} We still need to account for failure cases within the \texttt{privacy\_map} code. There are two places in which the code raises an exception:
+ 
+ \begin{enumerate}
+     \item \texttt{d\_in} is negative. The code does a check if \texttt{d\_in < 0} in line \ref{line:positivedin} and raises an error and terminates here if the statement is true. 
+     \item \texttt{inf\_div} fails. This step is only reached if \texttt{d\_in} is nonnegative. As defined in the pseudocode definitions doc, \texttt{inf\_div} throws an exception if division overflows from a 32-bit integer. 
+ \end{enumerate}
+ 
+ If none of these cases fail, \texttt{privacy\_map} returns \texttt{d\_in * inf\_div(scale)}.
+\end{enumerate}
```

Comment:
Do we need to explicitly consider failure cases in the privacy map? Failed evaluation of the privacy map should imply that it is false. The only difference with returning false is that you get richer diagnostic information.

Also, it returns `d_in.inf_div(scale)` (or `inf_div(d_in, scale)` if you prefer)